test/hoctests/tests/test_secref.py

```diff
@@ -0,0 +1,138 @@
+from neuron import h
+from neuron.expect_hocerr import expect_err, set_quiet
+
+set_quiet(0)
+
+# SectionRef unname and rename are undocumented but used in implementation
+# of CellBuild
+
+h("create a")
+sr = h.SectionRef(sec=h.a)
+assert sr.sec.name() == "a"
+assert sr.rename("b") == 0  # must first be unnamed
+sr.unname()
+expect_err("sr.unname()")  # already unnamed
+h("foo = 1")
+assert sr.rename("foo") == 0  # already exists
+sr.rename("b")
+assert sr.sec.name() == "b"
+h.delete_section(sec=h.b)
+assert sr.rename("c") == 0
+
+h("create a, b[3]")
+sr = h.SectionRef(sec=h.a)
+sr.unname()
+sr.rename("b")  # no longer an array and the old 3 are deleted
+assert sr.sec.name() == "b"
+h.delete_section(sec=h.b)
+
+# rename a bunch as an array
+h("create a, b, c, d")
+sr = None  # let's start over at 0
+sr = [h.SectionRef(sec=i) for i in h.allsec()]
+h.topology()
+print(sr)
+# the ones to rename into a new array
+lst = h.List()
+for i in [0, 2, 3]:
+    lst.append(sr[i])
+sr[0].unname()
+assert sr[0].rename("x", lst) == False  # have not unnamed c and d
```

Comment:
No strong opinions and I'm not sure what the current style guide guidance is, but I want to point out that here we're comparing to booleans... later we're seeing if things are booleans (via `is`)... one could also reasonable just say, e.g.
```python
assert not sr[0].rename("x", lst)
```
Is this inconsistency (using `==` vs `is`) on purpose?

Reply:
IIRC `is` compares identity of two object, while `==` compares equality.
https://stackoverflow.com/a/133024

It's somewhat annoying to get it to reveal the fact because:
```
a = 2
b = 2
```
end up pointing to the same object.

Reply:
I looked more closely at secref.cpp. It turns out rename and unname are returning double values 0.0 or 1.0.
I changed  the test to compare == to 0 or 1 instead of changing those secref.cpp functions to return False or True (via hoc_return_type_code = 2). Note that unname are rename are undocumented. The functions that do return a boolean are has_parent, has_trueparent, exists, and is_cas. Those now all compare using "is False" or "is True"